pkg/controller/ingressstate/ingress_state_controller.go

```diff
@@ -240,6 +240,9 @@ func checkEndpointHealthz(endpointIP string, rootCAs *x509.CertPool) error {
 	}
 	url := fmt.Sprintf("https://%s/healthz", net.JoinHostPort(endpointIP, "6443"))
 	response, err := client.Get(url)
+	if response != nil {
```

Comment:
compare with 

```
resp, err := http.Get("http://example.com/")
if err != nil {
	// handle error
}
defer resp.Body.Close()
```

from https://golang.org/pkg/net/http/. Let's not risk a nil `response.Body`.